src/portfft/committed_descriptor_impl.hpp

```diff
@@ -428,8 +423,13 @@ class committed_descriptor_impl {
       }
       return false;
     };
-    detail::factorize_input(fft_size, check_and_select_target_level);
-    return {detail::level::GLOBAL, param_vec};
+    if (!detail::factorize_input(fft_size, check_and_select_target_level)) {
+      param_vec.clear();
+      fft_size = static_cast<IdxGlobal>(std::pow(2, ceil(log(static_cast<double>(fft_size)) / log(2.0))));
+      detail::factorize_input(fft_size, check_and_select_target_level);
+      detail::factorize_input(fft_size, check_and_select_target_level);
```

Comment:
Why is this called twice?

Reply:
>I guess there is some state associated with check_and_select_target_level that means it does something different after each call?
This isn't immediately clear. Maybe a good start is explicitly defining the lambda captures?
Where possible, I find pure functions easier to follow.

Reply:
I don't like this because
1. It *really* looks like a bug.
2. It is hard to follow.

Would the following be accurate?
```suggestion
      // Forward DFT within Bluestein implementation (pre convolution).
      detail::factorize_input(fft_size, check_and_select_target_level);
      // Backward DFT within Bluestein implementation (post convolution).
      detail::factorize_input(fft_size, check_and_select_target_level);
```

Reply:
| There is no state associated with check_and_select_target_impl

`check_and_select_target_impl` modifies `param_vec`. When I mentioned pure functions, I wasn't trying to say there is anything wrong with lambdas, just unexpected side effects.